sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java

```diff
@@ -78,6 +83,9 @@ private static String readVersion() {
   private static Resource readResourceFromProviders() {
     Resource result = Resource.EMPTY;
     for (ResourceProvider resourceProvider : ServiceLoader.load(ResourceProvider.class)) {
+      if (RESOURCES_CONFIG.getDisabledResourceProviders().contains(resourceProvider.toString())) {
```

Comment:
I'm concerned with this relying on the `toString()` implementation of a given `ResourceProvider`.  In general, `toString` should be reserved for debugging purposes, rather than actual runtime functionality. If we need to have a way to identify resource provider implementations, we should add a method to the interface for this purpose. I recommend something like `getId()` or `getIdentifier()` or `getName()`.